worker/addfmusite/addFMUSite.py

```diff
@@ -0,0 +1,99 @@
+#  (1) Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+#  disclaimer.
+#
+#  (2) Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+#  disclaimer in the documentation and/or other materials provided with the distribution.
+#
+#  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+#  derived from this software without specific prior written permission from the respective party.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+#  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+#  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+#  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+#  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+#  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+########################################################################################################################
+
+from __future__ import print_function
+import sys
+import os
+import boto3
+import json
+import tarfile
+import shutil
+import time
+from subprocess import call
+import logging
+
+if len(sys.argv) == 3:
+    fmu_upload_name = sys.argv[1]
+    upload_id = sys.argv[2]
+else:
+    print('addSite called with incorrect number of arguments: %s.' % len(sys.argv), file=sys.stderr)
+    sys.exit(1)
+
+if not upload_id:
+    print('upload_id is empty', file=sys.stderr)
+    sys.exit(1)
+
+directory = os.path.join('/parse', upload_id)
+
+try:
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+except:
+    print('error making add site parsing directory for upload_id: %s' % upload_id, file=sys.stderr)
+    sys.exit(1)
+
+s3 = boto3.resource('s3', region_name='us-east-1', endpoint_url=os.environ['S3_URL'])
+key = "uploads/%s/%s" % (upload_id, fmu_upload_name)
+# fmu files gets uploaded with user defined names, but here we rename
+# to model.fmu to avoid keeping track of the (unreliable, non unique) user upload name
+# createFMUTags will however use the original fmu upload name for the display name of the site
+fmupath = os.path.join(directory, 'model.fmu')
+jsonpath = os.path.join(directory, 'tags.json')
+
+bucket = s3.Bucket('alfalfa')
+bucket.download_file(key, fmupath)
+
+call(['python', 'addfmusite/createFMUTags.py', fmupath, fmu_upload_name, jsonpath])
```

Comment:
All of the code before and after this line, is very similar to addSite.py which corresponds to the tagging for osm files. I wonder if we can factor this out to a common directory, and distinguish only by the create<FMU|osm>Tags.py that is called.

Reply:
This one is done!